torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -285,7 +287,7 @@ def unscale_(self, optimizer: torch.optim.Optimizer) -> None:
         found_inf_on_cudas = []
 
         for found_inf in optimizer_state["found_inf_per_device"].values():
-            if found_inf.device.type == "cpu":
+            if self._device == "cuda" and found_inf.device.type == "cpu":
                 found_inf_on_cpus.append(found_inf)
                 found_inf_on_cuda = found_inf.cuda()
                 found_inf_on_cudas.append(found_inf_on_cuda)
```

Comment:
This looks wrong

Reply:
Could you please help me understand? I am not very familiar with ddp.
The consideration here is to keep the original behavior on cuda. In addition, if it is a CPU GradScaler, there is no need to perform `found_inf_on_cuda = found_inf.cuda()`, but rather perform `dist.all_reduce(found_inf, async_op=True, group=self.process_group)` directly.